week-6/Homework/mandatory/1-alarmclock/alarmclock.js

```diff
@@ -1,4 +1,25 @@
-function setAlarm() {}
+function setAlarm() {
+  let timeValue = document.getElementById("alarmSet").value;
+  let minValue = Math.floor(timeValue / 60);
+  let secValue = timeValue % 60;
+  alarmSet = setInterval(() => {
+    if (secValue === 0 && minValue > 0) {
+      minValue--;
+      secValue = 59;
+    } else if (secValue > 0) {
+      secValue--;
+    } else {
+      playAlarm();
+    }
+    document.getElementById("timeRemaining").innerText =
+      "Time Remaining:" + addZero(minValue) + " : " + addZero(secValue);
+  }, 1000);
+}
+let alarmSet;
```

Comment:
the function works well
but it has one issue, it needs to clear interval each time I set the alarm, or it will keep going down